src/Webapi/Dom/Webapi__Dom__RectList.res

```diff
@@ -1,4 +1,4 @@
-type t = {
+type t = private {
```

Comment:
this is not really an objection, it just seems a little odd to be able to pattern match array_like stuff like this:

```rescript
let string = switch array {
  | {length: 0} => "zero"
  | {length} if length > 0 => "non-zero"
  | _ => "no match"
}
```

```js
var string = length !== 0 ? (
    length > 0 ? "non-zero" : "no match"
  ) : "zero";
```

seems like it's perfectly fine output from the playground though

Reply:
yes, records as objects opens up a lot of interesting pattern match possibilities. But they all compile to valid JS, even if it's a little weird, so I'm not worried.